Revit_Core_Engine/Query/Passes.cs

```diff
@@ -267,31 +268,38 @@ public static bool Passes(this Parameter parameter, FilterByParameterNumber requ
                 }
                 else
                     return false;
+            }
+            else
+            {
+                paramValue = double.NaN;
+            }
 
-                double comparisonValue = request.Value;
-                double comparisonTolerance = request.Tolerance;
+            double comparisonValue = request.Value;
+            double comparisonTolerance = request.Tolerance;
 
-                if (request.ConvertUnits)
-                {
-                    comparisonValue = comparisonValue.FromSI(parameter.Definition.GetSpecTypeId());
-                    comparisonTolerance = comparisonTolerance.FromSI(parameter.Definition.GetSpecTypeId());
-                }
+            if (request.ConvertUnits)
+            {
+                comparisonValue = comparisonValue.FromSI(parameter.Definition.GetSpecTypeId());
+                comparisonTolerance = comparisonTolerance.FromSI(parameter.Definition.GetSpecTypeId());
+            }
 
-                switch (request.NumberComparisonType)
-                {
-                    case NumberComparisonType.Equal:
-                        return Math.Abs(paramValue - comparisonValue) <= comparisonTolerance;
-                    case NumberComparisonType.Greater:
-                        return paramValue - comparisonValue > comparisonTolerance;
-                    case NumberComparisonType.GreaterOrEqual:
-                        return paramValue - comparisonValue > -comparisonTolerance;
-                    case NumberComparisonType.Less:
-                        return paramValue - comparisonValue < -comparisonTolerance;
-                    case NumberComparisonType.LessOrEqual:
-                        return paramValue - comparisonValue < comparisonTolerance;
-                    case NumberComparisonType.NotEqual:
-                        return Math.Abs(paramValue - comparisonValue) > comparisonTolerance;
-                }
+            switch (request.NumberComparisonType)
+            {
+                case NumberComparisonType.Equal:
+                    return Math.Abs(paramValue - comparisonValue) <= comparisonTolerance;
```

Comment:
One more edge case to cover:
```suggestion
                    return Math.Abs(paramValue - comparisonValue) <= comparisonTolerance || (double.IsNaN(paramValue) && double.IsNaN(comparisonValue));
```